unit.cpp

```diff
@@ -2529,7 +2529,7 @@ int Unit::GetAttribute(char const *attrib)
 					// Ignore mage only items for non-mages
 				} else if (ap->mods[index].flags & AttribModItem::PERMAN) {
 					int men = GetMen();
-					if (men <= items.GetNum(item))
+					if (men > 0 && men <= items.GetNum(item))
```

Comment:
My unit tests for the unit code actually found this.   I had create a unit with no men, and then was checking what stealth levels could see it, and discovered that it was being treated as having stealth 3.   That was due to the unit having 0 men, but this code checking if the men <= amount of items (RING).. since 0 <= 0, it was true and the unit got a stealth of 3!

Making sure that men >= 0 here fixes this bug and now the stealth is 0.   I added a unit test which asserts this as part of the suite after finding it :)

Reply:
oh this is an interesting discovery

Reply:
I thought so :) j